flycheck.el

```diff
@@ -5521,19 +5521,25 @@ function `buffer-file-name'."
 (defun flycheck-parse-xml-region (beg end)
   "Parse the xml region between BEG and END.
 
-Wrapper around `xml-parse-region' which transforms the return
-value of this function into one compatible to
+Try parsing with libxml first; if that fails, revert to
+`xml-parse-region'.  Failures can be caused by slightly incorrect
+XML (see URL `https://github.com/flycheck/flycheck/issues/1298'),
+or on Windows by a missing libxml DLL with a libxml-enabled Emacs
+build (see URL `https://github.com/flycheck/flycheck/issues/1330').
+
+The output of `xml-parse-region' is made compatible with that of
 `libxml-parse-xml-region' by simply returning the first element
-from the node list."
-  (car (xml-parse-region beg end)))
+of the node list."
+  ;; FIXME use `libxml-available-p' when it gets implemented.
+  (or (and (fboundp 'libxml-parse-xml-region)
+           (libxml-parse-xml-region beg end))
+      (ignore-errors (car (xml-parse-region beg end)))))
```

Comment:
Is this wise to silence errors here?

Reply:
Do you mean here vs somewhere else, or silencing vs not silencing? If the latter, I don't think we have much of a choice: that's how libxml behaves (it just returns nil when it fails to parse), which we then complain about (*unexpected output …*).  Otherwise we'd throw an error, and we don't currently do that.

Reply:
Okay, so now at least it's coherent with `libxml`.